src/take.js

```diff
@@ -0,0 +1,26 @@
+import { isRegExp, isString, isFunction } from 'lodash'
+
+const take = (match, callback) => (action, store) => {
+  const isMatching = ( // test matching
```

Comment:
The matcher should have it's own file, because if you want to create other listeners (like takeLatest...), you will reuse it. 

also, `src/take.js` may should be in `src/listeners/take.js`

But it can be done later, when you will have other listeners...

Reply:
Yes we will do that later, and we can override `take` like I did on `k-mst-onaction`: https://github.com/alakarteio/k-mst-onaction/blob/master/src/helpers/take.js#L27